fix(Checkbox): Remove empty label element

poi33 opened the pull request:
New styling that changed the label::before into input::before
Fixes #1740

poi33 commented:
This is an attempt to fix: https://github.com/digdir/designsystemet/issues/1735. 
Only the checkbox side.
Should be possible to apply the same changes to the radio input.

Barsnes commented:
Hi, thanks for the PR!
I'll let @mimarz look at this, but be is currently on vacation until next week 🐣

mimarz reviewed (changes requested):
Awesome!

First of thanks for wanting to contribute! 🤝 
 
There is a lot of good improvements here ❤️ 

Here is some general feedback.

1. Is it possible to solve this just using `.input` instead of `.input::before`? This might make it easier to solve next bulletpoints.
1. Focus styling looks broken. This is probably due to browsers native outline styling for `input` elements which needs to be overriden now.
![image](https://github.com/digdir/designsystemet/assets/1070981/600a3899-5c5e-41ca-8755-ed81ce0c4c80)
1. We aim for [Target Size AAA](https://www.w3.org/WAI/WCAG22/Understanding/target-size-enhanced.html) on our selection controls (radio, checkboxes, switches etc.) which is missing now. Maybe this can be solved using `:before` & `:after` to get that target size back (`2.75rem`)? 🤔 
![image](https://github.com/digdir/designsystemet/assets/1070981/1ecb26c0-2654-4956-b95f-a3691246baf6)

1. We use[ Github keywords](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) for linking our PRs. This needs to be used in the description of the PR for it to work. I made a separate task for just fixing this in Checkbox, so you can link it to #1740

poi33 commented on packages/react/src/components/form/Checkbox/Checkbox.module.css:
```diff
@@ -5,106 +5,113 @@
-  position: relative;
+  display: grid;
+  grid-template-columns: var(--fds-checkbox-size) auto;
+  gap: 6px 6px;
```
Should be fixed in e12068b

poi33 commented:
@mimarz added new commits

1. Changed to use input instead of input::before for the styling
2. Removed outline from input element
3. The input::before now is set to a 44px (2.75rem) size. (Had some trouble with the hover reflecting this)
4. Updated to use github keywords. (also added the git issue number to the commit header)

mimarz commented on packages/react/src/components/form/Checkbox/Checkbox.module.css:
```diff
@@ -18,93 +56,71 @@
+}
+
+.input:focus-visible {
   outline: var(--fds-checkbox-focus-border-width) solid var(--fds-semantic-border-focus-outline);
```
You probably need to also add `outline-offset: 0;` to get the correct focus style now with the styling changes to use `input`. 

In this PR:
![image](https://github.com/digdir/designsystemet/assets/1070981/45297f8a-81c8-431e-ac2f-24ee5307860f)

Design/Current:
![image](https://github.com/digdir/designsystemet/assets/1070981/6cef7750-5d2e-40bc-85ab-6619e72142d4)

poi33 replied:
<img width="150px" src="https://github.com/digdir/designsystemet/assets/3340349/35ed4565-2bdf-4252-8444-26b0f5051a28"/>
<img width="150px" src="https://github.com/digdir/designsystemet/assets/3340349/5ed53470-e96b-4b75-8bdd-a08e8c229dcb">

Forgot to test the state cross browser. 😞 
I'll add the fix now

mimarz replied:
ah, yeah, its easy to forget test other browsers 😅

mimarz reviewed (changes requested):
Some fine bit of work here! Found a small bug still with focus styling.

mimarz commented on packages/react/src/components/form/Checkbox/Checkbox.module.css:
```diff
@@ -116,19 +133,19 @@
   }
 
-  .container:not(.readonly, .disabled) > .input:hover:checked:focus-visible + .label::before {
+  .container:not(.readonly, .disabled) > .input:hover:checked:focus-visible::before {
```
This looks abit weird 😅 

![image](https://github.com/digdir/designsystemet/assets/1070981/fb0f4d2c-546c-4dd0-8e29-bed77dd7ca7f)

Looks like missed removing a `::before` when changing to styling on `input` element.

I think this suggestion should fix it 🤔 
```suggestion
  .container:not(.readonly, .disabled) > .input:hover:checked:focus-visible {
```

mimarz reviewed (approved):
Jolly good work sir! 
Thank your for your contribution, we appreciate it! ❤️ 🎉  

![welcome-to-the-society-of-uncompromising-men-cheers](https://github.com/digdir/designsystemet/assets/1070981/a9217213-7692-49dc-a98b-5c52b5cd537e)
